packages/interaction-kit/src/commands/options/channel.ts

```diff
@@ -1,13 +1,50 @@
+import type { APIApplicationCommandChannelOption } from "discord-api-types/v10";
 import { ApplicationCommandOptionType } from "discord-api-types/v10";
-import Option, { BaseOptionArgs } from "./option";
+import { isEqual } from "../../helpers/arrays";
+import type { Optional } from "../../interfaces";
+import type { BaseBasicOptionArgs } from "./option";
+import { BasicOption } from "./option";
 
-export default class ChannelOption extends Option {
-	constructor({ name, description, required }: BaseOptionArgs) {
+type ChannelOptionArgs = {
+	channelTypes?:
+		| APIApplicationCommandChannelOption["channel_types"]
+		| undefined;
+} & Optional<BaseBasicOptionArgs, "required">;
+
+export default class ChannelOption extends BasicOption<APIApplicationCommandChannelOption> {
+	public readonly channelTypes:
+		| APIApplicationCommandChannelOption["channel_types"]
+		| undefined;
+
+	constructor({
+		name,
+		description,
+		required,
+		channelTypes,
+	}: ChannelOptionArgs) {
 		super({
 			type: ApplicationCommandOptionType.Channel,
 			name,
 			description,
 			required,
 		});
+
+		this.channelTypes = channelTypes;
+	}
+
+	serialize() {
+		const payload = super.serialize();
+
+		if (this.channelTypes != null) {
```

Comment:
Well...what if someone explicitly sets it to null to reset it 👀 
Then what

Reply:
Would it not be an empty array?

Reply:
I don't know how your API works :P so I don't know if empty array acts like null or not

Reply:
Hm ok. This might be another case of needing the REST types - I'll leave a note in the issue but ignore it for now